pkg/ccl/changefeedccl/fetch_table_bytes_test.go

```diff
@@ -319,6 +320,39 @@ func TestFetchChangefeedUsageBytesE2EDisabledByDefault(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestFetchSizesUnredactedErrors(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	res := &roachpb.SpanStatsResponse{Errors: []string{"some", "stuff"}}
+
+	tss := mocks.NewMockTenantStatusServer(ctrl)
+	tss.EXPECT().SpanStats(gomock.Any(), gomock.Any()).Return(res, nil).MinTimes(1)
+
+	fx := newUsageFixtureWithMockTSS(ctx, t, tss)
+	defer fx.close()
+
+	fx.db.Exec(t, "CREATE TABLE testdb.test as SELECT generate_series(1, 1000) AS id")
+	row := fx.db.QueryRow(t, `CREATE CHANGEFEED FOR TABLE testdb.test INTO 'null://' WITH initial_scan='no';`)
+	var feedJobId int64
+	row.Scan(&feedJobId)
+
+	payload, err := fx.getChangefeedPayload(ctx, catpb.JobID(feedJobId))
+	require.NoError(t, err)
+	_, err = changefeedccl.FetchChangefeedUsageBytes(ctx, &fx.execCfg, payload)
+	require.Error(t, err)
+
+	err = errors.UnwrapAll(err)
+	errStr := strings.Join(err.(interface{ SafeDetails() []string }).SafeDetails(), ",") //nolint:errcmp
```

Comment:
if anyone knows of a better way to achieve this, pls lmk. afaict there's an error hierarchy like `withstack(stack, inner=leaferror(msg))`. SafeDetails() seems like the right method to use, but if i call it on the top level (withstack) error, i just get a stack trace as output.